pkg/utils/utils.go

```diff
@@ -53,12 +54,22 @@ func RetrieveVcConfigSecret(params map[string]interface{}, logger logrus.FieldLo
 
 	ns := "kube-system"
 	secretApis := clientset.CoreV1().Secrets(ns)
-	vsphere_secret := "vsphere-config-secret"
-	secret, err := secretApis.Get(vsphere_secret, metav1.GetOptions{})
+	vsphere_secrets := []string{"vsphere-config-secret", "csi-vsphere-config"}
+	var secret *k8sv1.Secret
+	for _, vsphere_secret := range vsphere_secrets {
+		secret, err = secretApis.Get(vsphere_secret, metav1.GetOptions{})
+		if err == nil {
+			logger.Infof("Retrieved k8s secret, %s", vsphere_secret)
+			break
+		}
+		logger.WithError(err).Infof("Failed to get k8s secret, %s", vsphere_secret)
+	}
+
+	// No valid secret found.
```

Comment:
Please add one more line of error log message on `if err != nil`. For example,

logger.Errorf("Failed to get any of the expected secrets, %v, for vSphere credential", vsphere_secrets)

It would give users more clear idea about what the error is.

Reply:
The reason I did not add an error message here is
- we already printed the error as an info message in the loop
- the caller of this routine is printing an error

But I can add the additional error message if you think it helps in debugging.

Reply:
It definitely would be helpful. The log level used in the for loop is INFO. We should explicitly tell users there is an error here and what the error is.